app/components/step-slider.jsx

```diff
@@ -0,0 +1,264 @@
+
+  // resizeHandler needs to access outerRef and setOuterRec but never change so that the event can be removed
+  // FTI resizeHandler gets called on initial render
+  const resizeHandler = useCallback(() => {
+    if (outerRef.current) {
+      let rect = outerRef.current.getBoundingClientRect()
+      if (rect.height && rect.width) {
+        // there is an issue on smartphones when rotating from landscape to portrait where the screen ends up shows a split between two components
+        // to work around this we are turning of transitions and then turning them back on after the viewport size stableizes
+        if (_this.timeout) clearTimeout(_this.timeout)
+        else setTransitions(false) // careful - the value of transitions will never be changed inside this memorized callback
+        _this.timeout = setTimeout(() => {
+          if (outerRef.current) {
+            // just to make sure
+            let rect = outerRef.current.getBoundingClientRect()
+            if (rect.height && rect.width) setOuterRect(rect)
+          }
+          setTransitions(true)
+          _this.timeout = 0
+        }, 100)
+        setOuterRect(rect)
+      }
+    }
+  }, [])
+
+  // if window resizes we need to recalculate or so the boxes are same size as new viewport
+  useEffect(() => {
+    window.addEventListener('resize', resizeHandler)
+    return () => window.removeEventListener('resize', resizeHandler)
+  }, [])
+
+  // if the other props have changed, we need to rerender the children
+  // _this.otherProps is only changed if it's shallow - different
+  // _this is written directcly because we don't want to cause another rerender - we just want to save the value for next time
+  if (!shallowEqual(_this.otherProps, otherProps)) _this.otherProps = otherProps
+
+  // has to be useLaoutEffect not useEffect or transitions will get enabled before the first render of the children and it will be blurry
+  if (typeof window !== 'undefined') {
+    useLayoutEffect(() => {
+      if (navRef.current) {
+        let navRect = navRef.current.getBoundingClientRect()
+        if (navRect.height && navRect.width) setNavBarRect(navRect)
+      }
+      if (footerRef.current) {
+        let footRect = footerRef.current.getBoundingClientRect()
+        if (footRect.height && footRect.width) setFooterRect(footRect)
+      }
+    }, [navRef.current, footerRef.current])
+  }
+  if (typeof window !== 'undefined') useLayoutEffect(resizeHandler, [outerRef.current])
+
+  function reducer(state, action) {
+    switch (action.type) {
+      case 'increment':
+        // Set next navigated-to panel as seen so we can render it
+        let currentStep = Math.min(state.currentStep + 1, children.length - 1)
+
+        if (steps) {
+          let newStatuses = {
+            ...stepStatuses,
+            [currentStep]: { ...stepStatuses[currentStep], seen: true },
+          }
+          setStepStatuses(Object.keys(newStatuses).map(key => newStatuses[key]))
```

Comment:
Instead of useState for stepStatuses, put it in state of this reducer.  Calling a hook (the userState hook) from within another hook (the userReducer) often gets crazy.  This may help with the problem you are having.

Reply:
Ah yeah, the statuses are consistent after switching. Looks like it's come full circle - as before I switched to useState because I couldn't figure why the Next button doesn't activate until I resize the window on Step 1, and doesn't activate at all on the next steps. Even `useEffect(() => {}, [state.stepStatuses])` isn't triggering until the resize.